pkgs/games/ldmud/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib, fetchFromGitHub, stdenv, autoreconfHook, pkg-config, bison, libiconv
+, pcreSupport ? true, pcre, gcryptSupport ? true, libgcrypt, erqSupport ? "erq"
+, compatModeSupport ? false, filenameSpacesSupport ? false, ipv6Support ? false
+, mccpSupport ? false, zlib, mysqlSupport ? false, libmysqlclient
+, postgresSupport ? false, postgresql, sqliteSupport ? false, sqlite
+, jsonSupport ? false, json_c, xmlSupport ? false, libxml2
+, deprecatedSupport ? true, tlsSupport ? false, openssl, pythonSupport ? false
+, python39, pythonSupportVersion ? python39 }:
+
+let supportedErq = [ "no" "erq" "xerq" ];
+in assert builtins.elem erqSupport supportedErq;
+
+stdenv.mkDerivation rec {
+  pname = "ldmud";
+  version = "3.6.6";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-2TaFt+T9B5Df6KWRQcbhY1E1D6NISb0oqLgyX47f5lI=";
+    name = pname;
```

Comment:
```suggestion
```

Reply:
It seems like the justification for this is below:

```
   # LD keeps its source code in a 'src' subfolder. Note the 'sourceRoot' is
   # relative to the tmp build directory nix creates so we need the predictable
   # 'name' set above.
   sourceRoot = "${pname}/src";
```

What's the more-correct pattern for getting the right sourceRoot? I see some packages using `sourceRoot = "source/src";`

Reply:
I mentioned this in the comment on `sourceRoot` below, it's necessary to set the `name` explicitly to `pname` in order for the `sourceRoot` to be correct. When I delete just this line I get an error during build:
```
error: builder for '/nix/store/glgi1v6fkg90ip1y556p0aa3bqy8k7f7-ldmud-3.6.6.drv' failed with exit code 1;
       last 4 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/qn3lv2qnflkp0qsqsakwq2hzcsdmb8ds-source
       > source root is ldmud/src
       > chmod: cannot access 'ldmud/src': No such file or directory
```

Similarly not setting `sourceRoot` or using `sourceRoot = "src";` doesn't work in my testing.

Reply:
>  I see some packages using sourceRoot = "source/src";

Ah! That seems workable. Thanks @abathur I'll go with that vs setting `name`.